src/OSX/Avalonia.MonoMac/TopLevelImpl.cs

```diff
@@ -144,6 +151,75 @@ public void SetCursor(NSCursor cursor)
                 UpdateCursor();
             }
 
+            public override NSDragOperation DraggingEntered(NSDraggingInfo sender)
```

Comment:
This logic shouldn't be in a platform-specific implementation. It should raise some events that are then handled by `TopLevel.cs.` Something like `public Action<RawDragEventArgs> OnDrag {get;set;}`